src/streaming/CaptionsLoader.js

```diff
@@ -0,0 +1,183 @@
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *  this list of conditions and the following disclaimer in the documentation and/or
+ *  other materials provided with the distribution.
+ *  * Neither the name of Dash Industry Forum nor the names of its
+ *  contributors may be used to endorse or promote products derived from this software
+ *  without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS AS IS AND ANY
+ *  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ *  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ *  INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ *  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ *  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+import Constants from './constants/Constants';
+import DashConstants from '../dash/constants/DashConstants';
+import URLLoader from './net/URLLoader';
+import URLUtils from './utils/URLUtils';
+import TextRequest from './vo/TextRequest';
+import DashJSError from './vo/DashJSError';
+import {HTTPRequest} from './vo/metrics/HTTPRequest';
+import EventBus from '../core/EventBus';
+import Events from '../core/events/Events';
+import Errors from '../core/errors/Errors';
+import FactoryMaker from '../core/FactoryMaker';
+import VTTParser from './utils/VTTParser';
+
+function CaptionsLoader(config) {
+
+    config = config || {};
+    const context = this.context;
+    const debug = config.debug;
+    const eventBus = EventBus(context).getInstance();
+    const urlUtils = URLUtils(context).getInstance();
+
+    let instance,
+        logger,
+        urlLoader,
+        parser;
+
+    let mssHandler = config.mssHandler;
+    let errHandler = config.errHandler;
+
+    function setup() {
+        logger = debug.getLogger(instance);
+
+        urlLoader = URLLoader(context).create({
+            errHandler: config.errHandler,
+            dashMetrics: config.dashMetrics,
+            mediaPlayerModel: config.mediaPlayerModel,
+            requestModifier: config.requestModifier,
+            useFetch: config.settings.get().streaming.lowLatencyEnabled,
+            urlUtils: urlUtils,
+            constants: Constants,
+            dashConstants: DashConstants,
+            errors: Errors
+        });
+    }
+
+    function createParser(data) {
+        if (data.indexOf('WEBVTT') > -1) {
+            return VTTParser(context).getInstance();
+        }
+        // } else if (data.indexOf('MPD') > -1 || data.indexOf('Patch') > -1) {
```

Comment:
Remove comments

Reply:
Done in https://github.com/Dash-Industry-Forum/dash.js/pull/3573/commits/97006dbd0ebd00b9b76ea0087921f0c54c418f8c